bundles/org.eclipse.ltk.core.refactoring/src/org/eclipse/ltk/core/refactoring/RefactoringDescriptorProxy.java

```diff
@@ -108,7 +108,7 @@ public final int hashCode() {
 		int code= getDescription().hashCode();
 		final long stamp= getTimeStamp();
 		if (stamp >= 0)
-			code+= (17 * stamp);
+			code+= 17 * Long.hashCode(stamp);
 		return code;
```

Comment:
The entire body of this method could be replaced by
`Objects.hash(getDescription(), getTimeStamp())`

Only disadvantage would be the Long object creation, but I assume that's something that is not relevant.